include/util/shared_memory_vector_wrapper.hpp

```diff
@@ -173,12 +175,19 @@ void swap(SharedMemoryWrapper<DataT> &lhs, SharedMemoryWrapper<DataT> &rhs) noex
     std::swap(lhs.m_size, rhs.m_size);
 }
 
-template <typename DataT, bool UseSharedMemory> struct ShM
+template <typename DataT, osrm::storage::MemorySetting MemorySetting> struct ShM
 {
-    using vector = typename std::conditional<UseSharedMemory,
-                                             SharedMemoryWrapper<DataT>,
-                                             std::vector<DataT>>::type;
+    using vector =
+        typename std::conditional<MemorySetting == osrm::storage::MemorySetting::SharedMemory,
+                                  SharedMemoryWrapper<DataT>,
+                                  std::vector<DataT>>::type;
 };
+
+template <typename DataT>
+using vector_view = typename util::ShM<DataT, osrm::storage::MemorySetting::SharedMemory>::vector;
+template <typename DataT>
+using InternalMemoryVector =
```

Comment:
I don't think there is a need for this type alias. This is nothing else then an `std::vector`. If we want to use that explicitly we can just write `std::vector`.